### ISSUE TITLE
finalize log streaming for pkg-executor

### ISSUE THREAD

**Issue by jbrockopp** [go-vela/community#368]
## Description

> NOTE: https://github.com/go-vela/community/issues/369 will likely need to be completed first

Part of the effort for https://github.com/go-vela/community/issues/156

Based off of https://github.com/go-vela/community/issues/349

Related to https://github.com/go-vela/community/pull/339

We need to finalize and complete the initial POC given for Option 2

The code for this POC can be found here:

https://github.com/go-vela/pkg-executor/compare/feature/log_streaming/opt_two?expand=1

## Value

Users are able to view logs for a `service`/`step` in near real-time

## Definition of Done

The code from the initial POC is accepted and merged into the [pkg-executor](https://github.com/go-vela/pkg-executor) codebase

## Effort  (Optional)

4 days

## Impacted Personas (Optional)

everyone
